testScripts/ObjectDownloadMedia.js

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable no-console */
+const ScriptBase = require("./parentClasses/ScriptBase");
+const utils = require("../src/Utils");
+const Fraction = require("fraction.js");
+const {JSONPath} = require("jsonpath-plus");
+const fs = require("fs");
+const path = require("path");
+
+class ObjectDownloadMedia extends ScriptBase {
+  async body() {
+    const client = await this.client();
+
+    let objectId = this.args.objectId;
+    const versionHash = this.args.objectHash;
+    const offeringType = this.args.offeringType || "default";
+    const startTime = this.args.startTime;
+    const endTime = this.args.endTime;
+    const out = this.args.out || "./out";
+
+    if(objectId === undefined){
+      if(versionHash===undefined){
+        throw Error("require object-id or object-hash to be provided");
+      }
+      const res = utils.DecodeVersionHash(versionHash);
+      objectId = res.objectId;
+    }
+
+    const dirPath = path.resolve(out);
+    if (!fs.existsSync(dirPath)) {
+      fs.mkdirSync(dirPath, { recursive: true });
+      console.log(`Directory created at ${dirPath}`);
+    } else {
+      throw Error(`Directory already exists at ${dirPath}`);
+    }
+
+
+    const sourcesJsonPath = "offerings." + offeringType + ".media_struct.streams.video.sources[*]";
+    const libraryId = await client.ContentObjectLibraryId({objectId, versionHash});
+    const totalDurationJsonPath = "offerings." + offeringType + ".media_struct.streams.video.duration";
+
+    // get object metadata
+    let metadata = await client.ContentObjectMetadata({
+      libraryId,
+      objectId
+    });
+
+    let sourcesMetadata = JSONPath({
+      json: metadata,
+      path: sourcesJsonPath,
+      wrap: false
+    });
+    if(sourcesMetadata === undefined) {
+      throw new Error("no matching offerings metadata found");
+    }
+
+    let totalDurationMetadata = JSONPath({
+      json: metadata,
+      path: totalDurationJsonPath,
+      wrap: false
+    });
+    if(totalDurationMetadata === undefined) {
+      throw new Error("no matching total duration metadata found");
+    }
+
+    const totalDurationTimeBase = new Fraction(totalDurationMetadata.time_base);
+    const totalDuration = new Fraction(totalDurationMetadata.ts).mul(totalDurationTimeBase);
+    if(startTime < 0 || endTime > totalDuration) {
+      throw new Error(`start or end time provided are not within range: ${totalDuration}s`);
+    }
+
```

Comment:
Additional check?

- if startTime >= endTime throw...

Reply:
possibly more checks,
- totalDuration < startTime < endTime, throw error
- startTime < totalDuration < endTime, throw warning but set endTime to the total Duration